server/graphql/v2/mutation/AgreementMutations.ts

```diff
@@ -1,132 +1,149 @@
 import { GraphQLNonNull } from 'graphql';
 import { GraphQLDateTime, GraphQLNonEmptyString } from 'graphql-scalars';
+import type { FileUpload } from 'graphql-upload/Upload.mjs';
 
 import twoFactorAuthLib from '../../../lib/two-factor-authentication';
+import { UploadedFile } from '../../../models';
 import AgreementModel from '../../../models/Agreement';
 import { checkRemoteUserCanUseHost } from '../../common/scope-check';
 import { Unauthorized } from '../../errors';
 import { AccountReferenceInput, fetchAccountWithReference } from '../input/AccountReferenceInput';
 import { AgreementReferenceInput, fetchAgreementWithReference } from '../input/AgreementReferenceInput';
 import { Agreement } from '../object/Agreement';
 
-const agreementMutations = {
-  addAgreement: {
-    type: new GraphQLNonNull(Agreement),
-    description: 'Add an agreement for the given host account. Scope: "host".',
-    args: {
-      title: {
-        type: new GraphQLNonNull(GraphQLNonEmptyString),
-        description: 'Agreement title.',
+export default new Promise(resolve => {
+  import('graphql-upload/GraphQLUpload.mjs').then(({ default: GraphQLUpload }) => {
+    resolve({
+      addAgreement: {
+        type: new GraphQLNonNull(Agreement),
+        description: 'Add an agreement for the given host account. Scope: "host".',
+        args: {
+          title: {
+            type: new GraphQLNonNull(GraphQLNonEmptyString),
+            description: 'Agreement title.',
+          },
+          expiresAt: {
+            type: GraphQLDateTime,
+            description: 'Optional date in which this agreement expires.',
+          },
+          host: {
+            type: new GraphQLNonNull(AccountReferenceInput),
+            description: 'Host where the agreement will be created.',
+          },
+          account: {
+            type: new GraphQLNonNull(AccountReferenceInput),
+            description: 'Account that is a party in this agreement',
+          },
+          attachment: {
+            type: GraphQLUpload,
+            description: 'Agreement attachment',
+          },
+        },
+        async resolve(_: void, args, req: express.Request): Promise<AgreementModel> {
+          checkRemoteUserCanUseHost(req);
+
+          const host = await fetchAccountWithReference(args.host, {
+            throwIfMissing: true,
+          });
+
+          if (!req.remoteUser.isAdminOfCollective(host)) {
+            throw new Unauthorized('Only host admins can create agreements');
+          }
+
+          await twoFactorAuthLib.enforceForAccount(req, host);
+
+          const account = await fetchAccountWithReference(args.account, {
+            throwIfMissing: true,
+          });
+
+          const attachment: FileUpload = args.attachment;
+          console.log(attachment);
+
+          let uploadedFile;
+          if (attachment) {
+            uploadedFile = await UploadedFile.uploadGraphQl(await attachment, 'AGREEMENT_ATTACHMENT', req.remoteUser);
+          }
```

Comment:
I don't really get what's going on with types here. According to https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/graphql-upload/Upload.d.mts, `FileUpload` is supposed to be an object. It's not typed as `Promise<FileUpload>`; yet we're `await`ing for it.

Reply:
Typing was wrong, the attachment argument is `Promise<FileUpload>`